src/Analyzers/CSharp/Tests/UseCompoundAssignment/UseCompoundAssignmentTests.cs

```diff
@@ -810,5 +810,135 @@ void M(int a)
     }
 }");
         }
+
+        [WorkItem(38054, "https://github.com/dotnet/roslyn/issues/38054")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestIncrement()
+        {
+            await TestInRegularAndScript1Async(
+@"public class C
+{
+    void M(int a)
+    {
+        a [||]= a + 1;
+    }
+}",
+@"public class C
+{
+    void M(int a)
+    {
+        a++;
+    }
+}");
+        }
+
+        [WorkItem(38054, "https://github.com/dotnet/roslyn/issues/38054")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestDecrement()
+        {
+            await TestInRegularAndScript1Async(
+@"public class C
+{
+    void M(int a)
+    {
+        a [||]= a - 1;
+    }
+}",
+@"public class C
+{
+    void M(int a)
+    {
+        a--;
+    }
+}");
+        }
+
+        [WorkItem(38054, "https://github.com/dotnet/roslyn/issues/38054")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestMinusIncrement()
+        {
+            await TestInRegularAndScript1Async(
+@"public class C
+{
+    void M(int a)
+    {
+        a [||]= a + (-1);
+    }
+}",
+@"public class C
+{
+    void M(int a)
+    {
+        a--;
+    }
+}");
+        }
+
+        [WorkItem(38054, "https://github.com/dotnet/roslyn/issues/38054")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestIncrementDouble()
+        {
+            await TestInRegularAndScript1Async(
+@"public class C
+{
+    void M(double a)
+    {
+        a [||]= a + 1.0;
+    }
+}",
+@"public class C
+{
+    void M(double a)
+    {
+        a++;
+    }
+}");
+        }
+
+        [WorkItem(38054, "https://github.com/dotnet/roslyn/issues/38054")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        public async Task TestIncrementNotOnString()
+        {
+            await TestInRegularAndScript1Async(
+@"public class C
+{
+    void M(string a)
+    {
+        a [||]= a + ""1"";
+    }
+}",
+@"public class C
+{
+    void M(string a)
+    {
+        a += ""1"";
+    }
+}");
+        }
+
+        [WorkItem(38054, "https://github.com/dotnet/roslyn/issues/38054")]
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)]
+        [InlineData("byte")]
+        [InlineData("short")]
+        [InlineData("long")]
+        [InlineData("float")]
+        public async Task TestIncrementLiteralConversion(string typeName)
+        {
+            await TestInRegularAndScript1Async(
+$@"public class C
+{{
+    void M({typeName} a)
+    {{
+        a [||]= a + ({typeName})1;
+    }}
+}}",
+$@"public class C
+{{
+    void M({typeName} a)
+    {{
+        a++;
+    }}
+}}");
+        }
```

Comment:
can you add a test with 'char'?

Reply:
Well, the operator `++` is defined on `char`, but `+` seems undefiend. If I'm correct, this codefix converts (valid)+ to ++, so it can't be triggered on `char`?

Reply:
i honestly don't care about which behavior youpick :)  i just would like a test that shows the current behavior.  the idea here is that this then allows us to completely  rewrite the impl and know if it's in line with how the current system works.  As this was a PR affecting behavior around builtin types/literals, i wanted the core set of numeric-like types covered for completeness :)

Reply:
Also, do you have a test for decimal?  i couldn't find one.

Reply:
I forgot enum, sorry.

For char, it should be impossible to trigger this diagnostic or write a valid test, because no `+` present.

Reply:
Well, interesting. The diagnostic doesn't depend on successful semantics. For char it should provide a code fix.

Reply:
> For char, it should be impossible to trigger this diagnostic or write a valid test, because no + present.

A test showing that it doesn't trigger on 'char' is a valid test :)